# HG changeset patch
# User Rik <rik@octave.org>
# Date 1478836982 28800
#      Thu Nov 10 20:03:02 2016 -0800
# Node ID dc828d9a7f3ab1e175199ea47ea88d08ad38fbaa
# Parent  4f144938eb65a96ff47a2ddc48ee9e99c078ab67
Fix building gl2ps when --without-opengl is used (bug #47886).

* gl2ps-print.cc: Add requirement for HAVE_OPENGL to the existing #if defined
for HAVE_GL2PS_H.
* gl2ps-print.cc (gl2ps_print): Add octave_unused_arg() in #else branch of ifdef
to suppress compilation warning messages about an unused input.

diff --git a/libinterp/corefcn/gl2ps-print.cc b/libinterp/corefcn/gl2ps-print.cc
--- a/libinterp/corefcn/gl2ps-print.cc
+++ b/libinterp/corefcn/gl2ps-print.cc
@@ -23,17 +23,17 @@ along with Octave; see the file COPYING.
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 // Both header files are required outside of HAVE_GLP2S_H
 #include "errwarn.h"
 #include "gl2ps-print.h"
 
-#if defined (HAVE_GL2PS_H)
+#if defined (HAVE_GL2PS_H) && defined (HAVE_OPENGL)
 
 #include <cstdio>
 
 #include <limits>
 
 #include <gl2ps.h>
 
 #include "lo-mappers.h"
@@ -797,17 +797,17 @@ namespace octave
 
   // If the name of the stream begins with '|', open a pipe to the command
   // named by the rest of the string.  Otherwise, write to the named file.
 
   void
   gl2ps_print (const graphics_object& fig, const std::string& stream,
                const std::string& term)
   {
-#if defined (HAVE_GL2PS_H)
+#if defined (HAVE_GL2PS_H) && defined (HAVE_OPENGL)
 
     // FIXME: should we have a way to create a file that begins with the
     // character '|'?
 
     bool have_cmd = stream.length () > 1 && stream[0] == '|';
 
     FILE *fp = 0;
 
@@ -841,13 +841,17 @@ namespace octave
     gl2ps_renderer rend (fp, term);
 
     rend.draw (fig, stream);
 
     // Make sure buffered commands are finished!!!
     rend.finish ();
 
 #else
+    octave_unused_parameter (fig);
+    octave_unused_parameter (stream);
+    octave_unused_parameter (term);
+
     err_disabled_feature ("gl2ps_print", "gl2ps");
 #endif
   }
 }
 
